FreeRTOS/Demo/CORTEX_R4_RM46_HERCULES_MCU_GCC/CMakeLists.txt

```diff
@@ -0,0 +1,194 @@
+cmake_minimum_required(VERSION 3.9)
+
+SET(CMAKE_CROSSCOMPILING "TRUE" CACHE STRING "Set Cross Compiling to true" FORCE)
+
+# Strip the default MacOSX flags that cause cross-compilations to fail.
+SET(CMAKE_OSX_DEPLOYMENT_TARGET "" CACHE STRING "Force unset of the deployment target for iOS" FORCE)
+SET(CMAKE_OSX_SYSROOT "" CACHE STRING "Force unset of the deployment target for iOS" FORCE)
+
+# Set the compiler before declaring the project for the test build
+SET(CMAKE_C_COMPILER "arm-none-eabi-gcc")
+SET(CMAKE_ASM_COMPILER "arm-none-eabi-gcc")
+
+# Set the system processor and name before declaring the project
+# Needs to be set here otherwise it will fail the test compilation
+SET(CMAKE_SYSTEM_NAME "Generic" CACHE STRING "Target system is a generic ARM Processor")
+SET(CMAKE_SYSTEM_PROCESSOR "armv7-r" CACHE STRING "Target system is an ARM7r Processor")
+
+# Set the ASM and C compilation flags
+SET(CMAKE_ASM_FLAGS "-mcpu=cortex-r4 -mfpu=vfpv3-d16 -Og -g -ggdb -Wall -MMD -MP")
+SET(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -specs=\"nosys.specs\" -specs=\"nano.specs\"")
+SET(CMAKE_C_FLAGS "${CMAKE_ASM_FLAGS} -marm -mfloat-abi=hard")
+
+project(RM46_FreeRTOS C ASM)
+cmake_minimum_required(VERSION 3.5)
+
+SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR} CACHE STRING "")
+
+# Increase the debug level of the CMAKE build
+SET(CMAKE_VERBOSE_MAKEFILE ON)
+
+# Get the absolute path to the Demo Directory
+SET(DEMO_DIR_REL "${CMAKE_CURRENT_SOURCE_DIR}")
+GET_FILENAME_COMPONENT(DEMO_DIR ${DEMO_DIR_REL} ABSOLUTE)
+
+# Get the absolute path to the Board Files
+SET(BOARD_FILES_DIR_REL "${DEMO_DIR}/BoardFiles")
+GET_FILENAME_COMPONENT(BOARD_FILES_DIR ${BOARD_FILES_DIR_REL} ABSOLUTE)
+
+SET(FREERTOS_CONFIG_FILE_DIRECTORY "${DEMO_DIR}/include" CACHE STRING "Config File Path")
+SET(FREERTOS_PORT "GCC_ARM_CRX_MPU" CACHE STRING "FreeRTOS Port to Use")
+
+# Clone the tag of the FreeRTOS-Kernel last tested with this project.
+INCLUDE(FetchContent)
+
+FetchContent_Declare(
+    FreeRTOS-Kernel
+    GIT_REPOSITORY https://github.com/Skptak/FreeRTOS-Kernel.git
+    GIT_TAG ARM_CRx_MPU
+    SOURCE_DIR "FreeRTOS-Kernel"
+    USES_TERMINAL_DOWNLOAD YES
+    USES_TERMINAL_UPDATE   YES
+    BUILD_COMMAND "")
+
+FetchContent_MakeAvailable(FreeRTOS-Kernel)
+
+# Get the absolute path to the FreeRTOS-Kernel Directory
+SET(FREERTOS_KERNEL_DIR_REL "${freertos-kernel_SOURCE_DIR}")
+GET_FILENAME_COMPONENT(FREERTOS_KERNEL_DIR ${FREERTOS_KERNEL_DIR_REL} ABSOLUTE)
+
+# Get the absolute path to the Port Directory
+SET(PORT_DIR_REL "${freertos-kernel_SOURCE_DIR}/portable/GCC/ARM_CRx_MPU")
+GET_FILENAME_COMPONENT(PORT_DIR ${PORT_DIR_REL} ABSOLUTE)
+
+# Debug
+MESSAGE("Project: ${PROJECT_NAME}")
+MESSAGE("Demo Directory: ${DEMO_DIR}")
+MESSAGE("FREERTOS_KERNEL_DIR: ${FREERTOS_KERNEL_DIR}")
+MESSAGE("PORT_DIR: ${PORT_DIR}")
+
+INCLUDE_DIRECTORIES(
```

Comment:
Can see how this makes sense to do. Bit loaded here since I'm sort of overloading the targets just to have different definitions for the demo configuration.

But if this was going to be split up from being this simple demo build system into something more substantial I'd agree that this should be done